lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -1054,29 +1054,29 @@ namespace Js
             scriptContext->ResetWeakReferenceDictionaryList();
             scriptContext->SetIsFinalized();
             scriptContext->GetThreadContext()->UnregisterScriptContext(scriptContext);
+            scriptContext->MarkForClose();
         }
     }
 
     void JavascriptLibraryBase::Dispose(bool isShutdown)
     {
         if (scriptContext)
         {
-            if (isShutdown)
-            {
-                // during shut down the global object might not be closed yet.
```

Comment:
these work was duplicate in ScriptContext::Close(true) which called from ~ScriptContext(), which will be triggered by the HeapDelete right after here